pkg/controller/sparkapplication/controller.go

```diff
@@ -476,14 +482,6 @@ func (c *Controller) syncSparkApplication(key string) error {
 			appToUpdate.Status.AppState.State = v1alpha1.PendingRerunState
 			appToUpdate.Status.AppState.ErrorMessage = ""
 		}
-	case v1alpha1.FailedSubmissionState:
-		if !shouldRetry(appToUpdate) {
-			// App will never be retried. Move to terminal FailedState.
-			appToUpdate.Status.AppState.State = v1alpha1.FailedState
-			c.recordSparkApplicationEvent(appToUpdate)
-		} else if hasRetryIntervalPassed(appToUpdate.Spec.RestartPolicy.OnSubmissionFailureRetryInterval, appToUpdate.Status.SubmissionAttempts, appToUpdate.Status.LastSubmissionAttemptTime) {
```

Comment:
We should remove OnSubmissionFailureRetryInterval since it's not used anywhere as Job doesn't take the retry interval as an input.

Reply:
Yes, removed.